draft-ietf-taps-arch.md

```diff
@@ -348,7 +348,25 @@ The Transport System Implementation Concepts define the set of objects used inte
+
+If two different Protocol Stacks can be safely swapped, or raced in parallel {{racing}}, then they are considered to be "equivalent". Equivalent Protocol Stacks must meet the following criteria:
+
+1. Both stacks must offer the same interface to the application for connection establishment and data transmission. For example, if one Protocol Stack has UDP as the top-level interface to the application, then it is not equivalent to a Protocol Stack that runs TCP as the top-level interface. Among other differences, the UDP stack would allow an application to read out message boundaries based on datagrams sent from the Remote Endpoint, whereas TCP does not preserve message boundaries on its own.
+
+2. Both stacks must offer the same transport services, as required by the application. For example, if an application specifies that it requires reliable transmission of data, then a Protocol Stack using UDP without any reliability layer on top would not be allowed to replace a Protocol Stack using TCP. However, if the application does not require reliability, then a Protocol Stack that adds unnecessary reliability would be allowed as an equivalent Protocol Stack.
+
+3. Both stacks must offer the same security properties. See the security protocol equivalence section below for futher discussion {{security-equivalence}}.
+
+### Transport Security Equivalence {#security-equivalence}
+
+The inclusion of transport security protocols {{I-D.ietf-taps-transport-security}} in a Protocol Stack adds extra restrictions to Protocol Stack equivalence. Security features, such as encryption, can vary in the level of protection they provide to an application's data depending on which algorithms are used, as well as how the algorithm is used within the protocol. Different protocols or protocol versions that use the same cryptographic algorithms should not be assumed to provide equivalent data protection.
+
+To ensure that security protocols are not incorrectly swapped, Transport Services systems should only automatically generate equivalent Protocol Stacks when the transport security protocols within the stacks are identical. For example, the same version of TLS running over two different transport protocol stacks may be considered equivalent.
```

Comment:
How about: 

... are identical. For safety and security reasons, implementations should consider protocols identical if they are of the same type and version. For example, the same version of TLS running over two different transport protocol stacks may be considered equivalent, whereas TLS 1.2 and TLS 1.3 should not be considered equivalent.

Reply:
Thanks! Updated, and added document reference.